src/tools/check_version.ts

```diff
@@ -0,0 +1,95 @@
+import { execSync } from 'child_process';
+import axios from 'axios';
+
+/**
+ * A function to make an API call to open VSX in order to get details about the latest release of redhat.ansible extension
+ * @param currentVersion string representing the current version against which information is required
+ * @returns version of the latest release
+ */
+function makeGetRequest() {
+  return new Promise(function (resolve, reject) {
+    axios
+      .get('https://open-vsx.org/api/redhat/ansible')
+      .catch(function (error) {
+        if (error.response) {
+          // Request made and server responded
+          console.log(error.response.status);
+          resolve(error.response.status);
+        } else if (error.request) {
+          // The request was made but no response was received
+          console.log(error.request);
+          reject(error.request);
+        } else {
+          // Something happened in setting up the request that triggered an Error
+          console.log('Error', error.message);
+          reject(error.message);
+        }
+      })
+      .then((response) => {
+        if (response) {
+          const result = response.data.version;
+          resolve(result);
+        }
+      });
+  });
+}
+
+/**
+ * A function that makes a call to the makeGetRequest and checks if the curren version of redhat.ansible is released in Open VSX or not
+ * @param currentVersion string representing the current version against which information is required
+ * @returns true if current version is released, false otherwise
+ */
+
+async function releasedInOpenVSX(currentVersion: string): Promise<boolean> {
+  const responseVersion = await makeGetRequest();
+
+  return true ? responseVersion === currentVersion : false;
+}
+
+/**
+ * A function that makes a call to the makeGetRequest and checks if the curren version of redhat.ansible is released in VSCode Marketplace or not
+ * @param currentVersion string representing the current version against which information is required
+ * @returns true if current version is released, false otherwise
+ */
+
+function releasedInVSCMarketplace(currentVersion: string): boolean {
+  // Grab the version of extension from vscode marketplace
+  const result = execSync(
+    'node ./node_modules/vsce/out/vsce show --json redhat.ansible'
+  ).toString();
+
+  const VSCMarketplaceVersion = JSON.parse(result).versions[0].version;
+
+  return true ? VSCMarketplaceVersion === currentVersion : false;
```

Comment:
Same here: what is this? Are you confusing it with Python's if-else?

Reply:
Same here. Corrected.